sympy/matrices/expressions/tests/test_matpow.py

```diff
@@ -93,3 +93,30 @@ def test_doit_nested_MatrixExpr():
     Y = ImmutableMatrix([[2, 3], [4, 5]])
     assert MatPow(MatMul(X, Y), 2).doit() == (X*Y)**2
     assert MatPow(MatAdd(X, Y), 2).doit() == (X + Y)**2
+
+
+def test_identity_power():
+    k = Identity(n)
+    assert MatPow(k, 4).doit()==k
+    assert MatPow(k, 1).doit() == k
+    assert MatPow(k, -3).doit() == k
+    assert MatPow(k, 0).doit() == k
+    l = Identity(3)
+    assert MatPow(l, 2).doit() == l
+    assert MatPow(l, -1).doit() == l
+    assert MatPow(l, 0).doit() == l
+    
```

Comment:
You have trailing spaces on this line. FYI you should run ``./bin/test sympy/utilities/tests/test_code_quality.py`` the output will tell you about the code quality issues.


Here's what I get

```
$ ./bin/test sympy/utilities/tests/test_code_quality.py
================================================================ test process starts =================================================================
executable:         /home/leosartaj/projects/sympy/my/bin/python  (2.7.10-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
random seed:        56687482
hash randomization: on (PYTHONHASHSEED=2022273946)

sympy/utilities/tests/test_code_quality.py[6] F.....                                                                                            [FAIL]

______________________________________________________________________________________________________________________________________________________
_______________________________________________ sympy/utilities/tests/test_code_quality.py:test_files ________________________________________________
  File "/home/leosartaj/projects/sympy/sympy/utilities/tests/test_code_quality.py", line 228, in test_files
    check_directory_tree(SYMPY_PATH, test, exclude)
  File "/home/leosartaj/projects/sympy/sympy/utilities/tests/test_code_quality.py", line 107, in check_directory_tree
    check_files(glob(join(root, pattern)), file_check, exclusions)
  File "/home/leosartaj/projects/sympy/sympy/utilities/tests/test_code_quality.py", line 123, in check_files
    file_check(fname)
  File "/home/leosartaj/projects/sympy/sympy/utilities/tests/test_code_quality.py", line 146, in test
    test_this_file(fname, test_file)
  File "/home/leosartaj/projects/sympy/sympy/utilities/tests/test_code_quality.py", line 169, in test_this_file
    assert False, message_space % (fname, idx + 1)
AssertionError: File contains trailing whitespace: /home/leosartaj/projects/sympy/sympy/matrices/expressions/tests/test_matpow.py, line 108.

================================================ tests finished: 5 passed, 1 failed, in 2.88 seconds =================================================
DO *NOT* COMMIT!
```